math/sieve_of_eratosthenes.cpp

```diff
@@ -37,29 +31,24 @@ void sieve(uint32_t N) {
 /**
  * This function prints out the primes to STDOUT
  */
-void print(uint32_t N) {
+void print(uint32_t N, bool *isprime) {
     for (uint32_t i = 1; i <= N; i++) {
```

Comment:
1 is neither prime nor composite.
I understand the original code has the erro. But since the code is being fixed, lets fix it for good and fix all problems that exist :)
`i` should start from `2`